### PR TITLE
Submission#51

### LINKED ISSUE #51: Submission component
```
<Submission
    className={}
    style={}
    electionOM={[electionObj,electionMethods]}
>
```
See #46 for electionOM details
```
electionObj={{...,submission: {link: url}, ...}
```

The inner frame of the submission box (the white box) fill show an iframe with the src being the link in the election obj.  If the link is not there, the moderator has not recorded yet.

Use this as the example of the link url for the story: https://cc.enciv.org/ucla-student-association-2021-moderator but we will try to end up with something that just shows the moderator. 

figma:  https://www.figma.com/file/yXrcMvg5EIwdLkHHxIdILh/Undebate-Self-Service-Portal?node-id=355%3A4667

- [x] See the for drawing below this one for the description of further functionality on the page

image

### PR DESCRIPTION
**Pull request by kamui-fin**
Closes #51

### CODE REVIEW

File: app/components/submission.js [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+import SvgCheck from '../svgr/check'
+import SvgReminder from '../svgr/reminder-sent'
+import SvgFeelingBlue from '../svgr/feeling-blue'
+
+export default function Submission(props) {
+    const { className, style, electionObj } = props
+    const classes = useStyles(props)
+
+    const getSubmission = () => {
+        const sortedSubmissions = electionObj.moderator.submissions.sort((a, b) => b.date - a.date)
```
Comment (ddfridley) [inline app/components/submission.js:16]: dates are strings, JS does not subtract strings well.  Use this 
```
list.sort(function (a, b) {
    return ('' + a.date).localeCompare(b.date);
})
```
But check the order you end up with. It might need to be reversed. 

ref: https://stackoverflow.com/questions/51165/how-to-sort-strings-in-javascript

File: stories/submission.stories.js [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    title: 'Submission',
+    component: Submission,
+    argTypes: {},
+}
+
+const Template = (args, context) => {
+    const { onDone } = context
+    return <Submission electionObj={onDone} {...args} />
+}
+
```
Comment (ddfridley) [inline stories/submission.stories.js:15]: Please add a test for Empty, where electionObj={}.  This will be what happens when the admin first creates a blank election.

File: stories/submission.stories.js [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+import React from 'react'
+import Submission from '../app/components/submission'
+
+export default {
+    title: 'Submission',
+    component: Submission,
+    argTypes: {},
+}
+
+const Template = (args, context) => {
```
Comment (ddfridley) [inline stories/submission.stories.js:11]: Please checkout [script.stories.js](https://github.com/EnCiv/undebate-ssp/blob/main/stories/script.stories.js) for how to pass electionOM to submission, rather than electionObj.

The story also shows how to initialize electionObj with defaultElectionObj.

File: app/components/submission.js [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+            </div>
+            <div className={classes.cornerPic}>{cornerPic}</div>
+        </div>
+    )
+}
+
+const useStyles = createUseStyles(theme => ({
+    container: {
+        padding: '2rem',
+        backgroundColor: '#ececec',
```
Comment (ddfridley) [inline app/components/submission.js:120]: I see this color in theme.js  backgroundColorApp
Since you are making changes, it's a good time to try the theme feature of react-css.
See app/components/candidate-table-input.js for an example using theme.js
I'm new to using themes too but I like having all these color definitions in one place.

**Review by ddfridley — CHANGES_REQUESTED** [review]
This is looking good, but needs a few changes.
Thanks!

**Review by ddfridley — APPROVED** [review]
